sbncode/FluxReader/FluxReaderAna_module.cc

```diff
@@ -0,0 +1,238 @@
+  // classes without bare pointers or other resource use.
+
+  // Plugins should not be copied or assigned.
+  FluxReaderAna(FluxReaderAna const&) = delete;
+  FluxReaderAna(FluxReaderAna&&) = delete;
+  FluxReaderAna& operator=(FluxReaderAna const&) = delete;
+  FluxReaderAna& operator=(FluxReaderAna&&) = delete;
+
+  // Required functions.
+  void analyze(art::Event const& e) override;
+
+private:
+
+  std::string _flux_label; ///< Label for flux dataproduct (to be set via fcl)
+  float _x_shift; // cm (to be set via fcl)
+  float _baseline; // cm (to be set via fcl)
+  float _nu_intersection_z; ///< Z position where to evaluate the flux (to be set via fcl)
+  float _nu_other_intersection_z; ///< Additional Z position where to evaluate the flux (to be set via fcl)
+
+  bool _apply_position_cuts; // wheter or not to apply position cuts (to be set via fcl)
+  float _x_cut; // cm, only saves neutrinos with x in [-_xcut, +_xcut] (to be set via fcl)
+  float _y_cut; // cm, only saves neutrinos with y in [-_ycut, +_ycut] (to be set via fcl)
+
+
+  const TDatabasePDG *_pdg_database = TDatabasePDG::Instance();
+
+  /// Returns the intersection point with the front face of the TPC
+  TVector3 GetIntersection(TVector3 nu_pos, TVector3 nu_dir, float z_location=0);
+
+  TTree* _tree;
+  bool _nu_hit; /// True if the neutrino hit the requested volumes
+  int _nu_pdg; /// PDG of neutrino
+  float _nu_e; /// Energy of the neutrino
+  float _nu_t; /// Time of the neutrino
+  float _nu_w; /// Neutrino weight
+  float _nu_x; /// X poisition of neutrino at the front face of the TPC
+  float _nu_y; /// Y poisition of neutrino at the front face of the TPC
+  float _nu_z; /// Z poisition of neutrino at the front face of the TPC
+  float _nu_px; /// X momentum of neutrino
+  float _nu_py; /// Y momentum of neutrino
+  float _nu_pz; /// Z momentum of neutrino
+  int _nu_decay; /// Neutrino parent decay code
+  float _nu_dk2gen; /// Distance from decay to ray origin
+  float _nu_p_angle; /// Angle between neutrino and parent direction
+  int _nu_p_type; /// Neutrino parent pdg
+  float _nu_p_dpx; /// Neutrino parent momentum x
+  float _nu_p_dpy; /// Neutrino parent momentum x
+  float _nu_p_dpz; /// Neutrino parent momentum x
+  float _nu_p_e; /// Neutrino parent energy
+  float _nu_r; /// Neutrino r
+  float _nu_oaa; /// Neutrino off axis angle
+
+  float _nu_other_x; /// X poisition of neutrino at the front face of the TPC (other)
+  float _nu_other_y; /// Y poisition of neutrino at the front face of the TPC (other)
+  float _nu_other_z; /// Z poisition of neutrino at the front face of the TPC (other)
+  float _nu_other_t; /// Time of the neutrino (other)
+};
+
+
+FluxReaderAna::FluxReaderAna(fhicl::ParameterSet const& p)
+  : EDAnalyzer{p}
+{
+
+  _flux_label = p.get<std::string>("FluxLabel", "flux");
+
+  _baseline = p.get<float>("Baseline", 11000); // cm
+  _x_shift = p.get<float>("XShift", -73.78); // cm
```

Comment:
Yes, good point. I added them a comment to explain them. Since these are SBND-specific numbers, I remove them from the defaults, and added them to the SBND flux reader fcl.